### PR TITLE
Add Support for int64

### PR DESCRIPTION
**Pull request by zhanghang1989**
## Description ##
Address the issue https://github.com/apache/incubator-mxnet/issues/9536

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by piiswrong** [comment]
Also need to add int8.
Could you turn
```
     .add_enum("float32", mshadow::kFloat32)
     .add_enum("float64", mshadow::kFloat64)
      .add_enum("float16", mshadow::kFloat16)
      .add_enum("uint8", mshadow::kUint8)
      .add_enum("int32", mshadow::kInt32)
     .add_enum("int64", mshadow::kInt64)
```
into a macro so we don't need to make the change every where each time?

**Comment by zhanghang1989** [comment]
@piiswrong added a macro.

File: src/operator/tensor/indexing_op.h [hunk 1]
```diff
@@ -836,12 +832,7 @@ struct OneHotParam : public dmlc::Parameter<OneHotParam> {
       .set_default(0.0f)
       .describe("The value assigned to the locations not represented by indices.");
     DMLC_DECLARE_FIELD(dtype)
-      .set_default(mshadow::kFloat32)
```
Comment (marcoabreu) [inline src/operator/tensor/indexing_op.h]: Was this removal intentional?

File: src/operator/mxnet_op.h [hunk 2]
```diff
@@ -204,6 +204,15 @@ inline int get_num_threads<cpu>(const int N) {
   }
 
 
+#define MXNET_ADD_ALL_TYPES \
+  .add_enum("float32", mshadow::kFloat32) \
+  .add_enum("float64", mshadow::kFloat64) \
+  .add_enum("float16", mshadow::kFloat16) \
+  .add_enum("uint8", mshadow::kUint8) \
```
Comment (piiswrong) [inline src/operator/mxnet_op.h]: also add int8

Comment (zhanghang1989) [reply src/operator/mxnet_op.h]: Thx. just added
